### ISSUE TITLE
[request] Migrate Bincrafters' icu_installer

### ISSUE THREAD

**Issue by Croydon** [conan-io/conan-center-index#1136]
### Package Details
  * Website: **http://site.icu-project.org/**
  * Source code: **https://github.com/unicode-org/icu/**
  * https://github.com/bincrafters/conan-icu

We already have a icu package for the libraries in CCI, but we can't archive the Bincrafters' variant just yet, because the CCI recipe doesn't package the binaries.

This leads continuously to confusions and puts us in a situation where we need to double maintenance work. Therefore I think it would be good when we fully migrate Bincrafters' ICU recipes rather sooner than later.

I'm unsure how to proceed here.

  * Do we add the binaries to the existing recipe?
  * Are we creating a new recipe for cross-compiling reasons or similar reasons?
    * If so, how do we name it? In this case the existing package should have probably be named `libicu` or something, but probably too late now?

**Comment by madebr** [comment]
Same resolution as tcl/tcl_installer? https://github.com/conan-io/conan-center-index/pull/848#issuecomment-585285857
Make the change when a new version of icu is released.

@jgsogo Are installer packages still needed when cross building lands in conan?

**Comment by jgsogo** [comment]
No, from `1.24` onwards (https://github.com/conan-io/conan/pull/5594) (probably we should wait some time to see if any issue arises), settings `os_build` and `arch_build` should be deprecated and packages for installers and libraries should be the same package (yes, `protobuf` and `protoc` should be only one package). The graph will expand two nodes with different settings for those packages if they are used in the build_machine (`protoc` executable) and also in the host_machine (library `protobuf`).

I assume that `tcl`/`tcl_installer` is the same situation as `protobuf`/`protoc` and any other installer.

**Comment by Croydon** [comment]
If this is already on the horizon we could move ahead and add binaries into the same package as libs. Merging later would mean deprecating a new package which is much more painful IMHO. 

Also I would really love to see protobuf/protoc unblocked. For CCI we could merge the protoc binary into the protobuf package. People who need cross-compiling support right now can still use the Bincrafters' recipes. This would actually mean no change compared to right now.

**Comment by jgsogo** [comment]
From my perspective, I would block all the _xxxx_installers_, the current model with `os/arch_build` is essentially broken and lead to lots of issues and workarounds... but I'm not the police in this repository, you all are doing much more work than myself here 🙌  Thanks!

**Comment by madebr** [comment]
@jgsogo 
If I'm correct, then the following changes are required to our recipes:
- `os_build`/`arch_build` must be changed to `os`/`arch`
- installer packages should only add the `bin` folder to `cpp_info`. Ideally, `env_info.PATH` should not be modified

Am I forgetting something?

**Comment by jgsogo** [comment]
Yes, but still this feature is in an early stage, I really think it is the way to go (and if breaking were an option, I would force always the new behavior and take the `build` profile equal to `host` if missing, or to the `default` one) but we really need more feedback about it. Today we may think that only the `bin` folder is needed, but we cannot change all the recipes to match that approach before testing it, maybe we realize that we need the `PATH` too, or something else...

Also, changing these _installer_ recipes in Conan will require a change in the CI, we would need to provide the `profile:build` too  while compiling in order to work with this new build requirements... and this is something we need to talk about for ConanCenter CI in the following weeks.

So, keep calm, try it before making breaking changes to the recipes. Give feedback, maybe we are wrong and this is not the right approach (it is 😎 )

**Comment by SpaceIm** [comment]
It's fixed now, isn't it?

**Comment by danimtb** [comment]
I think so: https://github.com/conan-io/conan-center-index/blob/master/recipes/icu/all/conanfile.py
Closing this then! Thank you

**Comment by Croydon** [comment]
If `icu` does contain now all executables as `icu_installer` in Bincrafters does, then yes

**Comment by Croydon** [comment]
I quickly compared two Windows packages of icu_installer/64.2 and icu/64.2. The Bincrafters packages has `uconv.exe` which the CCI packages does not have

**Comment by danimtb** [comment]
@Croydon please open a new issue to improve that in the recipe

**Comment by Croydon** [comment]
I will leave that to people who are actually missing this executable in the package 🙂
